EnvelopeItem/Topic.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Enqueue\MessengerAdapter\EnvelopeItem;
+
+use Symfony\Component\Messenger\EnvelopeItemInterface;
+
+/**
+ * @author Thomas Prelot <tprelot@gmail.com>
+ *
+ * @experimental in 4.1
+ */
+final class Topic implements EnvelopeItemInterface
```

Comment:
I'd actually name it more generic; `TransportConfiguration`? Which would contain a `topic` string field.